### PR TITLE
Make sure the issuer of the ID Token is the correct one.

### PR DESCRIPTION
**Pull request by rohe**
Should not accept an ID Token from anyone but the expected one.

### CODE REVIEW

**Comment by rohe** [comment]
Travis fails because #374 isn't accepted yet.

File: src/oic/oic/message.py [hunk 1]
```diff
@@ -674,6 +675,13 @@ class IdToken(OpenIDSchema):
     def verify(self, **kwargs):
         super(IdToken, self).verify(**kwargs)
 
+        try:
+            kwargs['iss'] == self['iss']
```
Comment (schlenk) [inline src/oic/oic/message.py]: Why should this produce an AssertionError? no assert anywhere...

It is also not a great idea to use 'assert' for this kind of code, as 'assert' gets removed when someone runs python with optimization (-O), so logic fails.

Comment (rohe) [inline src/oic/oic/message.py]: Correct, I should have know better.

**Comment by lwm** [comment]
:beer:

**Pull request merged**